File: Serene/Serene.Script/Imports/FormContexts/Northwind.CustomerDemographicForm.cs

namespace Serene.Northwind
{
    using Serenity;
    using Serenity.ComponentModel;
    using System;
    using System.Runtime.CompilerServices;

    public partial class CustomerDemographicForm : PrefixedContext
    {
        [InlineConstant] public const string FormKey = "Northwind.CustomerDemographic";
    
        public CustomerDemographicForm(string idPrefix) : base(idPrefix) {}
    
        public StringEditor CustomerTypeID { [InlineCode("{this}.w('CustomerTypeID', Serenity.StringEditor)")] get; private set; }
        public StringEditor CustomerDesc { [InlineCode("{this}.w('CustomerDesc', Serenity.StringEditor)")] get; private set; }
    }
}

File: Serene/Serene.Web/Modules/Common/Imports/ServerTypings/Northwind.CustomerDemographicForm.ts
namespace Serene.Northwind {
    export class CustomerDemographicForm extends Serenity.PrefixedContext {
        static formKey = 'Northwind.CustomerDemographic';

    }

    export interface CustomerDemographicForm {
        CustomerTypeID: Serenity.StringEditor;
        CustomerDesc: Serenity.StringEditor;
    }

    [['CustomerTypeID', Serenity.StringEditor], ['CustomerDesc', Serenity.StringEditor]].forEach(x => Object.defineProperty(CustomerDemographicForm.prototype, <string>x[0], { get: function () { return this.w(x[0], x[1]); }, enumerable: true, configurable: true }));
}

